src/mission_companion.cpp

```diff
@@ -476,11 +476,35 @@ bool talk_function::display_and_choose_opts(
         }
 
         int name_offset = 0;
-        calcStartPos( name_offset, sel, info_height, folded_names_lines );
+        size_t sel_pos = 0;
+        // Translate from entry index to line index
+        for (size_t i = 0; i < sel; i++) {
+            sel_pos += folded_names[i].size();
+        }
+
+        calcStartPos( name_offset, sel_pos, info_height, folded_names_lines );
+
+        int name_index = 0;
+        // Are we so far down the list that we bump into the end?
+        bool last_section = folded_names_lines < info_height || folded_names_lines - info_height <= size_t(name_offset);
+
+        // Translate back from desired line index to the corresponding entry, making sure to round up near the end to ensure the last line gets included.
+        if (name_offset > 0) {
+            for (size_t i = 0; i < cur_key_list.size(); i++) {
+                name_offset -= folded_names[i].size();
+                if (name_offset <= 0) {
+                    if (last_section && name_offset < 0) {
+                        name_index++;
+                    }
+                    break;
+                }
+                name_index++;
+            }
+        }
 
         size_t list_line = 2;
-        for( size_t current = name_offset; list_line < info_height &&
-             current < cur_key_list.size(); current++ ) {
+        for (size_t current = name_index; list_line < info_height &&
```

Comment:
```suggestion
        for( size_t current = name_index; list_line < info_height &&
```
CDDA C++ style is parentheses are not spaced after `for` or `if` and there is always a space after a `(` and before a `)`.

I'm not saying that it's a pretty style, just that it is the style.

Reply:
This one is going to be very painful as Astyle refuses to point out each and every instance. I very strongly recommend trying to get the style checker to provide feedback on what it rejects: you shouldn't have to waste time on correcting style issues.

It doesn't exactly help that the style is completely backwards for me, so I have to consciously check every single instance to adjust the style (not a complaint as such, but more of a heads up that I'm unlikely to be able to adjust to the style without a lot of slip ups, and probably never completely, as it conflicts with other styles used elsewhere).

Reply:
You can (and should) be running astyle through a tool. You can do it with `make astyle` if you're using make, and if you're not there's probably other ways mentioned in the documentation.